packages/fetchai/skills/confirmation_aw1/registration_db.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains a the registration db model."""
+
+import logging
+import os
+import sqlite3
+from typing import Tuple
+
+from aea.skills.base import Model
+
+
+_default_logger = logging.getLogger(
+    "aea.packages.fetchai.skills.carpark_detection.detection_database"
+)
+
+
+class RegistrationDB(Model):
+    """Communicate between the database and the python objects."""
+
+    def __init__(self, **kwargs):
+        """Initialise the Detection Database Communication class."""
+        super().__init__(**kwargs)
+        this_dir = os.getcwd()
+        self.db_path = os.path.join(this_dir, "registration.db")
+        self._initialise_backend()
+
+    def _initialise_backend(self) -> None:
+        """Set up database and initialise the tables."""
+        if os.path.isfile(self.db_path):
+            return
+        self._execute_single_sql(
+            "CREATE TABLE IF NOT EXISTS registered_table (address TEXT, ethereum_address TEXT, "
+            "ethereum_signature TEXT, fetchai_signature TEXT, "
+            "developer_handle TEXT, tweet TEXT)"
+        )
```

Comment:
you could consider using some field as primary key, so it's guaranteed unique insertion for that field.

https://www.sqlitetutorial.net/sqlite-primary-key/